packages/rockets-next-web/src/app/(main)/users/UsersTable.tsx

```diff
@@ -0,0 +1,103 @@
+"use client";
+
+import type { RowProps } from "@concepta/react-material-ui/dist/components/Table/types";
+import { type FC, useCallback, useMemo } from "react";
+import Box from "@mui/material/Box";
+import IconButton from "@mui/material/IconButton";
+import EditIcon from "@mui/icons-material/Edit";
+import ChevronRightIcon from "@mui/icons-material/ChevronRight";
+import DeleteIcon from "@mui/icons-material/Delete";
+
+import Table from "@/components/Table/Table";
+
+import { headers } from "./tableConfig";
+import type { FormData, ActionType } from "./types";
+
+interface UsersTableProps {
+  isLoading?: boolean;
+  isEmptyStateVisible?: boolean;
+  data: FormData[];
+  onActionClick: ({
+    rowData,
+    action,
+  }: {
+    rowData: FormData;
+    action: ActionType;
+  }) => void;
+}
+
+const UsersTable: FC<UsersTableProps> = ({
+  isLoading,
+  isEmptyStateVisible,
+  data,
+  onActionClick,
+}) => {
+  const getRowDataById = useCallback(
+    (rowId: FormData["id"]) => {
+      return data.find((item: FormData) => item.id === rowId);
+    },
+    [data]
+  );
+
+  const handleActionButtonClick = useCallback(
+    (rowId: FormData["id"], action: ActionType) => {
+      if (!getRowDataById(rowId)) {
+        return;
+      }
```

Comment:
you can pass the full object here, instead of just the id, in this case you don't need to make the find

Reply:
also here you are executing the getRowDataById twice, this is bad for performance

Reply:
Done! `getRowDataById` wasn't even necessary